README.md

```diff
@@ -1,285 +1,366 @@
-	 - _`uint buyAmt`_ - The amount that will be bought
-	 - _[TokenInterface](https://github.com/makerdao/oasis-direct-proxy/blob/master/src/OasisDirectProxy.sol#L11) `payToken`_ - An address of any `ERC-20 token` or token that implements the interface
-	 - _`uint maxPayAmt`_ - please check _*threshold_
+### threshold
 
-	**Returns:**
-	- _`uint payAmt`_ -the amount of `payToken` that will be paid
+In order to understand exactly what `threshold` does, let's go though a quick example. 
 
-	**Errors:**
-	- _`otc.getPayAmount`_ - might cause the TX to fail if there aren't enough order to fill the `buyAmt`
-	- might cause the TX to fail if the calculated amount to be paid is higher than `maxPayAmt`
-	- _`payToken.transferFrom`_ - might cause the TX to fail
-	- _`otc.buyAllAmount`_ - might cause the TX to fail if we have to pay more than the calculated amount to be paid in the beginning of the TX.
-	- _`buyToken.transfer`_ - might cause the TX to fail
+**Example:** Let's say that someone wants to sell `10 MKR` at the price of `1.5 MKR/DAI`. 
 
-	---
+Suppose the `buyAmt` will be `150 DAI`. By the time the transaction is executed, a [slippage](https://www.investopedia.com/terms/s/slippage.asp) may occur. This means that the price might have dropped to `1.2 MKR/DAI` and the caller will receive `120 DAI` instead of the exact uint `wethAmt` that was expected ( `150 DAI`). In order to mitigate such scenarios, we introduced **threshold** (`minBuyAmt` or `minPayAmt` ). 
 
-- **`buyAllAmountPayEth`** - Used to exchange `ETH` for `ERC-20 token`. This methods should be called with `ETH`. In the beginning the sent `ETH` is locked in `WETH token`. After that the `otc` contract is ensured to have enough allowance in the `wethToken` to do transfers on the behalf of `this` contract. Follows buy operation where `wethAmt` is calculated. The bought amount is transferred to the caller. Last step include sending the difference between the amount of `ETH` sent to the contract method and the actual amount needed to buy the `buyToken` amount.
-**Disclaimer:** It's important to pay attention that there is neither `minBuyAmt` nor `maxPayAmt`. The caller of the method should take into account the slippage that might occur and send more `ETH` ( the max amount he is willing to pay in order to get the `buyAmt`. If less `ETH` is spent, the remaining is returned to the caller.
+The value provided here determines the lowest possible price that allows the trade to go through and close. If the prices drop below the threshold, the transaction fails and everything is reverted. Following the example above, one can provide a value (in base units) such as `140 DAI`. This will allow the transaction to pass and then the trade will be deemed successful if the price of `MKR/DAI` does not drop below `1.4 MKR/DAI`.
```

Comment:
Well in general the threshold is not the price but an amount for one of the sides. We don't store the price anywhere but it's calculate.